source/distributed/tria_base.cc

```diff
@@ -147,9 +146,61 @@ namespace parallel
   TriangulationBase<dim, spacedim>::n_locally_owned_active_cells_per_processor()
     const
   {
+    Assert(
+      (number_cache.n_locally_owned_active_cells.size() > 0),
+      ExcMessage(
+        "You first have to call update_n_locally_owned_active_cells_per_processor!"));
+
     return number_cache.n_locally_owned_active_cells;
   }
 
+  template <int dim, int spacedim>
+  void
+  TriangulationBase<dim, spacedim>::
+    update_n_locally_owned_active_cells_per_processor()
+  {
+#ifdef DEAL_II_WITH_MPI
+    if (number_cache.n_locally_owned_active_cells.size() == 0)
```

Comment:
Is the size of this field reset to zero upon mesh refinement? Otherwise, you'd skip the computation if the field was still around from the previous refinement if that function had been called there.